### ISSUE TITLE
Incorrect types when using i18next in an esm package with moduleResolution node16

### ISSUE THREAD

**Issue by weareoutman** [i18next/i18next#1900]
## To Reproduce

https://codesandbox.io/s/vibrant-hellman-fgs4r9?file=/src/index.ts

```ts
import i18n from "i18next";

console.log(
  // This is actually undefined when running in browser or node
  typeof i18n.default,
  typeof i18n.default?.language,
  // ts error
  i18n.language
);
```

tsconfig.json:
```json
{
  "compilerOptions": {
    "strict": true,
    "module": "esnext",
    "esModuleInterop": true,
    "rootDir": "src",
    "moduleResolution": "node16"
  }
}
```

Note: change `moduleResolution` to `node` will make it work as expected.

package.json (It's ES Module):
```json
{
  "type": "module"
}
```

<img width="399" alt="image" src="https://user-images.githubusercontent.com/2338946/212034435-b3a7a87a-e458-4d44-a644-667f2193f675.png">

## Expected behavior

TS should not report an error for `i18n.language` , and instead, it should for `i18n.default`.

```js
// Paste the expected results here
```

## Your Environment

- *runtime version*: node v16
- *i18next version*: 22.4.9
- *os*: Mac
- *typescript version*: 4.9.4

**Comment by weareoutman** [comment]
Any thoughts?

**Comment by weareoutman** [comment]
This seems to be fixed in TypeScript 5.0.0-beta with `--moduleResolution bundler`

**Comment by pedrodurek** [comment]
I'll be closing this for now since it can be addressed with typescript 5.0
